### PR TITLE
Make navbar responsive

### PR DESCRIPTION
**Pull request by CosmicCoder96**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] The acceptance, integration and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:

Makes Navbar responsive
#### Changes proposed in this pull request:

- It makes the navbar responsive, earlier it overflowed on smaller screens.

image

**Fixes**: #8

### CODE REVIEW

File: app/components/nav-bar.js [hunk 1]
```diff
@@ -10,3 +10,11 @@ export default Component.extend({
     }
   }
 });
+
+$(document).ready(function (){
```
Comment (niranjan94) [inline app/components/nav-bar.js]: No. This is not how you're supposed to do things in Ember. The component may not always be ready by the time the document is. Move this inside `didInsertElement` method of the component class.

File: app/components/nav-bar.js [hunk 2]
```diff
@@ -10,3 +10,11 @@ export default Component.extend({
...
+	$('.right.menu.open').on("click",function(e){
```
Comment (niranjan94) [inline app/components/nav-bar.js]: Use ember actions

File: app/components/nav-bar.js [hunk 3]
```diff
@@ -10,3 +11,11 @@ export default Component.extend({
     }
   }
 });
+
+$(document).ready(function() {
```
Comment (niranjan94) [inline app/components/nav-bar.js]: No. This is not how you're supposed to do things in Ember. The component may not always be ready by the time the document is. Move this inside didInsertElement method of the component class.

File: app/components/nav-bar.js [hunk 4]
```diff
@@ -10,3 +11,11 @@ export default Component.extend({
...
+  $('.right.menu.open').on('click', function(e) {
```
Comment (niranjan94) [inline app/components/nav-bar.js]: Use ember actions and not click listeners.

**Comment by niranjan94** [comment]
> Instead of sidebar menu, bottom up menu for smaller sceen is used, to avoid issues with requirement of pusher class to be their on entire content and the navbar and rest of the content being direct children of body

@CosmicCoder96 a pusher is needed only if we want the sidebar to push the content to the right while opening. Instead the sidebar can come _over_ the content. For that, a pusher is not needed. Please do it that way.

File: app/components/nav-bar.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import Ember from 'ember';
+import $ from 'jquery';
```
Comment (niranjan94) [inline app/components/nav-bar.js]: Don't import jquery directly. Use `this.$('<selector>')` inside a component to access a component-level jquery object.

**Comment by CosmicCoder96** [comment]
> Also, when adding a sidebar, move it to a separate component to not clutter up the nav-bar component.
@niranjan94 Meaning, I create a new `.hbs` for sidebar and use that with the `nav-bar`.hbs?

**Comment by niranjan94** [comment]
@CosmicCoder96 no ... I mean this ... https://guides.emberjs.com/v2.12.0/components/defining-a-component/

This would be a good chance for you to get familiar with Ember.js and the project.

**Comment by niranjan94** [comment]
@CosmicCoder96 in fact, even the pushing style sidebar is possible without the need for it to be a direct parent of body.

**Comment by CosmicCoder96** [comment]
@niranjan94
When I did that it generated an error in the console, it said that they should be direct elements. Though it did move the content of the body and navbar with appropriate pusher class amended to it.
I will go through the documentation again for the same. thank you.

**Comment by niranjan94** [comment]
@CosmicCoder96 make use of the `context` parameter in semantic sidebar options to workaround that

**Comment by niranjan94** [comment]
@CosmicCoder96 if it's too new to you and you find it hard, lemme know. We can work on this issue together on this same branch. That way you can see how things can be done in Ember & semantic ui

**Comment by niranjan94** [comment]
Also, btw, just remembered. We already have a pusher here `templates/application.hbs` which you can re-use for the sidebar too.

**Comment by CosmicCoder96** [comment]
@niranjan94 I have made all the changes as you requested. The only issue which remains is I can't get the `context` right. (It's commented out in PR) The sidebar toggle works the first time. But for the rest of time it doesn't as semantic ui moves the navbar somewhere else and it is no longer a part of `this.$('.ui.sidebar.menu')` 

Here is a preview : 
image

**Comment by niranjan94** [comment]
@CosmicCoder96 have added a commit to this. Go through it and let me know if there's any part of it that is unclear. 😄

File: app/components/nav-bar.js [hunk 6]
```diff
@@ -7,10 +7,6 @@ export default Component.extend({
     logout() {
       this.get('authManager').logout();
       this.get('routing').transitionTo('index');
-    },
-    expand() {
-      // this.$('.ui.sidebar.menu').sidebar({context:this.$('')});
-      this.$('.ui.sidebar.menu').sidebar('show');
```
Comment (niranjan94) [inline app/components/nav-bar.js]: Not needed. We can use side-bars `attachEvent` method.

File: app/components/side-bar.js [hunk 7]
```diff
@@ -2,5 +2,15 @@ import Ember from 'ember';
 
 const { Component } = Ember;
 
-export default Ember.Component.extend({
+export default Component.extend({
+  didInsertElement() {
+    this._super.call(this);
+    this.$('.ui.sidebar')
+      .sidebar({ context: this.$() })
```
Comment (niranjan94) [inline app/components/side-bar.js]: Context is the div holding sidebar

File: app/styles/partials/navbar.scss [hunk 8]
```diff
@@ -1,8 +0,0 @@
-@media (max-width: 767px) {
```
Comment (niranjan94) [inline app/styles/partials/navbar.scss]: Use semantic-ui's own responsive helpers and the custom helpers already added to the project

File: app/styles/partials/footer.scss [hunk 9]
```diff
@@ -1,4 +1,4 @@
-.pusher {
+.footer-pusher {
```
Comment (niranjan94) [inline app/styles/partials/footer.scss]: Just renaming it to not get confused

File: app/templates/components/side-bar.hbs [hunk 10]
```diff
@@ -1,42 +1,43 @@
...
+    <a href="{{href-to 'login'}}" class="item">{{t 'Login'}}</a>
+  {{/if}}
+  {{#unless (eq routing.currentRouteName 'create')}}
+    <div class="item">
+      <a class="ui blue button" style="text-transform: uppercase; font-weight: 400" href="{{href-to 'create'}}">{{t 'Create Event'}}</a>
+    </div>
+  {{/unless}}
+</div>
+<div class="pusher">
+  {{yield}}
```
Comment (niranjan94) [inline app/templates/components/side-bar.hbs]: The rest of the body content within `{{#side-bar}} {{/side-bar}}` will be rendered inside this.

File: app/components/side-bar.js [hunk 11]
```diff
@@ -2,5 +2,15 @@ import Ember from 'ember';
...
+      .sidebar('setting', 'transition', 'push')
+      .sidebar('attach events', '.ui.sidebar .item')
+      .sidebar('attach events', '.ui.sidebar .link')
```
Comment (niranjan94) [inline app/components/side-bar.js]: Clicking on any of these, will toggle the sidebar

File: app/templates/application.hbs [hunk 12]
```diff
@@ -1,12 +1,15 @@
-<div class="pusher">
-  {{nav-bar}}
-  {{#if (eq routing.currentRouteName 'index')}}
-    {{welcome-header}}
-  {{/if}}
-  <div id="main-container" class="ui container main-container {{unless device.isInternetExplorer 'flexed-container' ''}}">
-    {{outlet}}
+{{#side-bar}}
+  <div class="footer-pusher">
```
Comment (CosmicCoder96) [inline app/templates/application.hbs]: @niranjan94 This <div class="footer-pusher"> was moved here to ensure that side bar's pusher content will include the the original navbar with the rest of the body?

Comment (niranjan94) [inline app/templates/application.hbs]: Everything inside `{{#side-bar}} {{/side-bar}}` will go inside the `<div class="pusher"> </div>` defined in `app/templates/components/side-bar.hbs`

**Comment by CosmicCoder96** [comment]
@niranjan94 Thank a lot for the detailed explanation of the changes. It's clear to me now 👍 .

**Pull request merged**